core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -429,13 +442,17 @@ private void process(Block block)
                         valueSet = null;
                         blockBuilder = null;
                     }
+                    else {
+                        retainedSizeInBytes = valueSet.getRetainedSizeInBytes();
+                    }
                     break;
                 case MIN_MAX:
                     updateMinMaxValues(block, minMaxComparison);
                     break;
                 case NONE:
                     break;
             }
+            return retainedSizeInBytes;
```

Comment:
Should we also account for `min`/`max` for completeness? Also should we nullify `min`/`max` when the domain is built?

Reply:
I've added changes to nullify min/max when domain is built.
Accounting for min/max feels a bit overkill right now. We're using `block.getSingleValueBlock`, that ensures that we'll hold on to a single value without holding on to the entire block. We'd have to be dealing with very long length strings for this to matter, those are also unlikely to be join keys. Still if we need to, it could be done in follow-ups.